internal/service/glue/schema.go

```diff
@@ -238,12 +238,12 @@ func resourceSchemaUpdate(d *schema.ResourceData, meta interface{}) error {
 			SchemaDefinition: aws.String(d.Get("schema_definition").(string)),
 		}
 
-		_, err := conn.RegisterSchemaVersion(defInput)
+		schemaVersion, err := conn.RegisterSchemaVersion(defInput)
 		if err != nil {
 			return fmt.Errorf("error updating Glue Schema Definition (%s): %w", d.Id(), err)
 		}
 
-		_, err = waitSchemaVersionAvailable(conn, d.Id())
+		_, err = waitSchemaVersionAvailable(conn, *schemaVersion.SchemaVersionId)
```

Comment:
This is the fix, previously this method would fetch the latest schema version. In the case of schema registration failure, the latest schema will return the latest `AVAILABLE` schema, not the newly created failed schema version.